src/Symfony/Component/Security/Core/Authentication/Token/UserAuthorizationCheckerToken.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authentication\Token;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * UserAuthorizationCheckerToken implements a token used for checking authorization.
+ *
+ * @author Nate Wiebe <nate@northern.co>
+ */
+class UserAuthorizationCheckerToken extends AbstractToken
+{
+    /**
+     * @param string[] $roles
+     */
+    public function __construct(UserInterface $user, array $roles = [])
+    {
+        parent::__construct($roles);
+
+        $this->setUser($user);
+    }
```

Comment:
Why does the constructor have a `$roles` property? We always use the user's roles anyway.

```suggestion
    public function __construct(UserInterface $user)
    {
        parent::__construct($user->getRoles());

        $this->setUser($user);
    }
```

Reply:
As to why, it was to line up closer to the other tokens, but it makes more sense to just grab from the user instead. Will be updated with the next push.